data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3ScanPartitionCreationSupplier.java

```diff
@@ -101,7 +101,7 @@ private List<PartitionIdentifier> listFilteredS3ObjectsForBucket(final List<Stri
                                                                      final LocalDateTime endDateTime,
                                                                      final Map<String, Object> globalStateMap) {
 
-        Instant mostRecentLastModifiedTimestamp = globalStateMap.containsKey(bucket) ? Instant.parse((String) globalStateMap.get(bucket)) : null;
+        Instant mostRecentLastModifiedTimestamp = globalStateMap.containsKey(bucket) && Objects.nonNull(globalStateMap.get(bucket)) ? Instant.parse((String) globalStateMap.get(bucket)) : null;
```

Comment:
`globalStateMap.containsKey(bucket) && Objects.nonNull(globalStateMap.get(bucket)) ` can simplify to `globalStateMap.get(bucket) != null`

Also, `Objects.nonNull` is not necessary here. It is intended for use where a `Predicate` is needed.